aicsimageio/readers/reader.py

```diff
@@ -834,6 +834,287 @@ def mosaic_tile_dims(self) -> Optional[Dimensions]:
+        select_scenes: Optional[
+                Union[List[Union[str, int]], Tuple[Union[str, int], ...]]]
+            Which scenes to stack into a single array. Scenes can be provided
+            as a list or tuple of scene indices or names. It is recommended to
+            use the scene integer index instead of the scene name to avoid
+            duplicate scene name lookup issues.
+            Default: None (stack all scenes)
+        scene_character: str
+            Character to use as the name of the scene dimension on the output
+            array. Default "S"
+
+        Returns
+        -------
+        stack: types.MetaArrayLike
+            The fully stacked array. This can be 6+ dimensions with Scene being
+            the first dimension.
+
+        """
+        mode_check = ["data", "dask_data", "xarray_data", "xarray_dask_data"]
+        if mode not in mode_check:
+            raise ValueError(
+                f"Invalid mode kwarg. Found {mode} but should be one of:"
+                f"{', '.join(mode_check)}."
+            )
+        scene_stacks = []
+        scene_names = []
+
+        if select_scenes is None:
+            select_scenes = list(range(len(self.scenes)))
+
+        for i, s in enumerate(select_scenes):
+            self.set_scene(s)
+            data = getattr(self, mode)
+
+            # Store the shape and dtype of the first scenes data
+            # to check against later scenes. If returning a DataArray
+            # store the coords and dims to use in the final output
+            if i == 0:
+                shape = data.shape
+                dtype = data.dtype
+
+                if "xarray" in mode:
+                    coords = dict(data.coords)
+                    if scene_character in coords:
+                        _ = coords.pop(scene_character)
+                    dims = data.dims
+
+            # Check other scenes against the first scene
+            else:
+                if data.shape != shape:
+                    if not drop_non_matching_scenes:
+                        raise exceptions.UnexpectedShapeError(
+                            f"All scenes must have same shape. Found shape"
+                            f"{data.shape} in scene {s} but expected"
+                            f"{shape} based on scene {select_scenes[0]}"
+                        )
+                    else:
+                        continue
+                if data.dtype != dtype:
+                    if not drop_non_matching_scenes:
+                        raise TypeError(
+                            f"All scenes must have the same dtype. Found data"
+                            f"with dtype {data.dtype} in scene {s} but expected"
+                            f"dtype {dtype} based on scene {select_scenes[0]}"
+                        )
+                    else:
+                        continue
+
+            scene_stacks.append(data)
+            scene_names.append(self.current_scene)
```

Comment:
I think the param should be "scene_coord_values='names'" vs "scene_coord_values='indices'"?

And then the coords become either ["Image0", "Image1", "Blah"] vs [0, 1, 2].

Reply:
Sounds good. One more question though: say the user provides `scene_coord_values='indices'` and`select_scenes = [1,3,5]`. Do we set the coords as [1,3,5] or [0,1,2]? 

My gut feeling is that in this case we should use [1,3,5] and but there should be a third option that is like `None` or `'reindex'` or something better that just doesn't attach any coords to the scene dimension so that xarray falls back to numpy like indexing. I also probably think this third case should be the default?

Reply:
Oh i figured it would always reindex in the integer form...